contracts/verifiers/ValidatorWhitelist.sol

```diff
@@ -51,9 +51,7 @@ contract ValidatorWhitelist is ZKPVerifierBase {
         uint64[] calldata requestIds,
         ZKPRequest[] calldata requests
     ) public virtual override {
-        for (uint256 i = 0; i < requestIds.length; i++) {
-            setZKPRequest(requestIds[i], requests[i]);
```

Comment:
We need to call `setZKPRequest` in ValidatorWhitelist instead of `super.setZKPRequest` to check `onlyWhitelistedValidator(request.validator)` for these requests.
So this can't be optimized like this.